src/spikeinterface/extractors/nwbextractors.py

```diff
@@ -176,92 +162,100 @@ def __init__(
 
-        num_frames = int(self._es.data.shape[0])
-
         # Extractors channel groups must be integers, but Nwb electrodes group_name can be strings
-        if "group_name" in self._nwbfile.electrodes.colnames:
-            unique_grp_names = list(np.unique(self._nwbfile.electrodes["group_name"][:]))
+        if "group_name" in electrodes_table.colnames:
+            unique_grp_names = list(np.unique(electrodes_table["group_name"][:]))
 
         # Fill channel properties dictionary from electrodes table
-        if "channel_name" in self._nwbfile.electrodes.colnames:
-            channel_ids = [self._es.electrodes["channel_name"][i] for i in self._es.electrodes.data]
+        if "channel_name" in electrodes_table.colnames:
+            channel_ids = [electrical_series.electrodes["channel_name"][i] for i in electrodes_indices]
         else:
-            channel_ids = [self._es.electrodes.table.id[x] for x in self._es.electrodes.data]
+            channel_ids = [electrical_series.electrodes.table.id[x] for x in electrodes_indices]
 
-        dtype = self._es.data.dtype
+        dtype = electrical_series.data.dtype
         BaseRecording.__init__(self, channel_ids=channel_ids, sampling_frequency=sampling_frequency, dtype=dtype)
+        num_frames = int(electrical_series.data.shape[0])
         recording_segment = NwbRecordingSegment(
             nwbfile=self._nwbfile,
-            electrical_series_name=self._electrical_series_name,
+            electrical_series_name=electrical_series_name,
             num_frames=num_frames,
             times_kwargs=times_kwargs,
         )
         self.add_recording_segment(recording_segment)
 
-        self.extra_requirements.extend(["pandas", "pynwb", "hdmf"])
+        #################
+        # Extract gains and offsets TODO: Should be a function
+        #################
 
         # Channels gains - for RecordingExtractor, these are values to cast traces to uV
-        gains = self._es.conversion * 1e6
-        if self._es.channel_conversion is not None:
-            gains = self._es.conversion * self._es.channel_conversion[:] * 1e6
+        gains = electrical_series.conversion * 1e6
+        if electrical_series.channel_conversion is not None:
+            gains = electrical_series.conversion * electrical_series.channel_conversion[:] * 1e6
 
         # Set gains
         self.set_channel_gains(gains)
 
+
         # Set offsets
-        offset = self._es.offset if hasattr(self._es, "offset") else 0
-        if offset == 0 and "offset" in self._nwbfile.electrodes:
-            electrode_table_index = self._es.electrodes.data[:]
-            offset = self._nwbfile.electrodes["offset"].data[electrode_table_index]
+        offset = electrical_series.offset if hasattr(electrical_series, "offset") else 0
+        if offset == 0 and "offset" in electrodes_table:
+            offset = electrodes_table["offset"].data[electrodes_indices]
```

Comment:
Just planning ahead, how much longer do we want to support the round-about way of storing offsets?

Reply:
Hmm.. I guess there might always be some historical files on DANDI (not sure off the top of my head) if SI health check on DANDI files is the concern... I guess always keep it then?

Reply:
Yes, I think this was the concern. Previous files created by us when we used to write the offset there.